crates/orchestrator/src/data_storage/aws_s3/mod.rs

```diff
@@ -22,25 +22,56 @@ pub struct AWSS3 {
 impl AWSS3 {
     /// Initializes a new AWS S3 client by passing the config
     /// and returning it.
-    #[allow(dead_code)]
-    pub async fn new(config: AWSS3Config) -> Self {
-        // AWS cred building
-        let credentials = Credentials::new(
-            config.s3_key_id.clone(),
-            config.s3_key_secret.clone(),
-            None,
-            None,
-            "loaded_from_custom_env",
-        );
-        let region = Region::new(config.s3_bucket_region.clone().to_string());
-        let conf_builder = Builder::new().region(region).credentials_provider(credentials);
-        let conf = conf_builder.build();
+    pub async fn new(config: AWSS3ConfigType) -> Self {
+        let (config_builder, config) = match config {
+            AWSS3ConfigType::WithoutEndpoint(config) => {
+                let credentials = Credentials::new(
+                    config.s3_key_id.clone(),
+                    config.s3_key_secret.clone(),
+                    None,
+                    None,
+                    "loaded_from_custom_env",
+                );
+                let region = Region::new(config.s3_bucket_region.clone().to_string());
```

Comment:
duplicate code, could be moved to its own internal function `get_credentials_and_region_from_config(config: &Config)`

Reply:
Added the function.